### PR TITLE
EVG-2215-populate-running-task-document

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: rest/route/host_routes.go [hunk 1]
```diff
@@ -110,14 +110,25 @@ func (high *hostIDGetHandler) Execute(ctx context.Context, sc data.Connector) (R
...
-	if err != nil {
+	if err = hostModel.BuildFromService(*foundHost); err != nil {
 		if _, ok := err.(*rest.APIError); !ok {
 			err = errors.Wrap(err, "API model error")
 		}
 		return ResponseData{}, err
 	}
 
+	if foundHost.RunningTask != "" {
+		runningTask, err := sc.FindTaskByID(foundHost.RunningTask)
```
Comment (john-m-liu) [inline rest/route/host_routes.go]: Correct casing is FindTaskById

File: rest/route/host_routes.go [hunk 2]
```diff
@@ -110,14 +110,25 @@ func (high *hostIDGetHandler) Execute(ctx context.Context, sc data.Connector) (R
...
+		if apiErr, ok := err.(*rest.APIError); !ok ||
+			(ok && apiErr.StatusCode != http.StatusNotFound) {
+			return []model.Model{}, nil, errors.Wrap(err, "Database error")
```
Comment (john-m-liu) [inline rest/route/host_routes.go]: These returns need to be wrapped in a ResponseData struct

**Comment by tychoish** [comment]
updated

File: rest/route/host_routes.go [hunk 3]
```diff
@@ -110,14 +110,25 @@ func (high *hostIDGetHandler) Execute(ctx context.Context, sc data.Connector) (R
...
 			err = errors.Wrap(err, "API model error")
 		}
 		return ResponseData{}, err
 	}
 
+	if foundHost.RunningTask != "" {
+		runningTask, err := sc.FindTaskById(foundHost.RunningTask)
+		if apiErr, ok := err.(*rest.APIError); !ok ||
+			(ok && apiErr.StatusCode != http.StatusNotFound) {
+			return ResponseData{}, nil, errors.Wrap(err, "Database error")
```
Comment (john-m-liu) [inline rest/route/host_routes.go]: this returns too many things

**Comment by tychoish** [comment]
fixed

**Pull request merged**